examples/fabric-admin/scripts/fabric-sync-app.py

```diff
@@ -0,0 +1,251 @@
+    if secured_device_port is not None:
+        args.extend(["--secured-device-port", str(secured_device_port)])
+    return await run("[FS-BRIDGE]", program, *args,
+                     stdin=asyncio.subprocess.DEVNULL)
+
+
+async def main(args):
+
+    if args.commissioner_node_id == 1:
+        raise ValueError("NodeID=1 is reserved for the local fabric-bridge")
+
+    storage_dir = args.storage_dir
+    if storage_dir is not None:
+        os.makedirs(storage_dir, exist_ok=True)
+    else:
+        storage = TemporaryDirectory(prefix="fabric-sync-app")
+        storage_dir = storage.name
+
+    pipe = args.stdin_pipe
+    if pipe and not os.path.exists(pipe):
+        os.mkfifo(pipe)
+
+    def terminate(signum, frame):
+        admin.terminate()
+        bridge.terminate()
+        sys.exit(0)
+
+    signal.signal(signal.SIGINT, terminate)
+    signal.signal(signal.SIGTERM, terminate)
+
+    admin, bridge = await asyncio.gather(
+        run_admin(
+            args.app_admin,
+            storage_dir=storage_dir,
+            rpc_admin_port=args.app_admin_rpc_port,
+            rpc_bridge_port=args.app_bridge_rpc_port,
+            paa_trust_store_path=args.paa_trust_store_path,
+            commissioner_name=args.commissioner_name,
+            commissioner_node_id=args.commissioner_node_id,
+            commissioner_vendor_id=args.commissioner_vendor_id,
+        ),
+        run_bridge(
+            args.app_bridge,
+            storage_dir=storage_dir,
+            rpc_admin_port=args.app_admin_rpc_port,
+            rpc_bridge_port=args.app_bridge_rpc_port,
+            secured_device_port=args.secured_device_port,
+            discriminator=args.discriminator,
+            passcode=args.passcode,
+        ))
+
+    # Wait a bit for apps to start.
+    await asyncio.sleep(1)
+
+    try:
+        # Check whether the bridge is already commissioned. If it is,
+        # we will get the response, otherwise we will hit timeout.
+        cmd = "descriptor read device-type-list 1 1 --timeout 1"
+        admin.stdin.write((cmd + "\n").encode())
+        await asyncio.wait_for(BRIDGE_COMMISSIONED.wait(), timeout=1.5)
+    except asyncio.TimeoutError:
+        # Commission the bridge to the admin.
+        cmd = "fabricsync add-local-bridge 1"
+        if args.passcode is not None:
+            cmd += f" --setup-pin-code {args.passcode}"
+        if args.secured_device_port is not None:
+            cmd += f" --local-port {args.secured_device_port}"
+        admin.stdin.write((cmd + "\n").encode())
+        # Wait for the bridge to be commissioned.
+        await asyncio.sleep(5)
+
+    # Open commissioning window with original setup code for the bridge,
+    # so it can be added by the TH fabric.
+    cmd = "pairing open-commissioning-window 1 0 0 600 1000 0"
+    admin.stdin.write((cmd + "\n").encode())
+    # Wait some time for the bridge to open the commissioning window.
+    await asyncio.sleep(1)
```

Comment:
is there a log we can wait on? It seems like we want a pattern of creating some wait object, issuing a command and waiting with timeout. Some pseudo-python would look like:

```py
log_wait = WaitForString('Reading attribute: Cluster=0x0000_001D Endpoint=0x1 AttributeId=0x0000_0000')
admin.SendCommand('descriptor read device-type-list 1 1 --timeout 1')
log_wait.wait(timeout_sec=1)
```


unsure a bit about what interfaces  are the cleanest, however it does seem like we would want an easy way to "send command" (I should not need to add \n and encode every time) and then wait for some output.